tests/Unit/DataStructures/Test_DataVector.cpp

```diff
@@ -171,6 +171,223 @@ void check_vectors(const T1& t1, const T2& t2) {
 }
 }  // namespace
 
+SPECTRE_TEST_CASE("Unit.DataStructures.DataVector.Math",
```

Comment:
I simply cannot review this. We need to formally verify that every change is a no-op even with regard to the potential non-owningness of DataVector.